include/libobmcsession/obmcsession.hpp

```diff
@@ -0,0 +1,77 @@
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#pragma once
+
+#include <libobmcsession/obmcsession.h>
+
+#include <libobmcsession/obmcsession_proto.hpp>
+#include <sdbusplus/bus.hpp>
+
+/** @brief Constructs session manager
+ *
+ * @param[in] bus           - Handle to system dbus
+ * @param[in] objPath       - The Dbus service slug uniquely identifying the
+ *                            source of a session to create the appropriate
+ *                            items from. Service name template:
+ *                            'xyz.openbmc_project.Session.${slug}'.
+ * @param[in] type          - Type of all session items that will be created
+ * by the current instance.
+ * @return int              - session manager initialization status
+ */
+extern int obmcsesManagerInitAsio(sdbusplus::bus::bus& bus,
+                                  const std::string& slug, uint8_t type);
+
+/**
+ * @brief Create a session and publish into the dbus without appropriate
+ *        session payload
+ *
+ * @param[in] cleanupFn             - the callback to cleanup session on
+ *                                    destroy.
+ * @param[out] SessionIdentifier    - unique session ID
+ *
+ * @note The session is incomplete. Metadata of session must be set for 10
+ *       seconds. Otherwise, the current session will be closed. All other
+ *       new session creates requests will be rejected until the current
+ *       created session accepts metadata or is removed by timeout.
+ *
+ * @return int                      - session build status
+ */
+extern int obmcsesCreateTransactionWithFCleanup(SessionCleanupFn&& cleanupFn,
+                                                obmcsesSessionId* sessionId);
+
+/**
+ * @brief Create a session and publish into the dbus with cleanup callback
+ *        on the session destroy.
+ *
+ * @param[in] userName              - the owner user name
+ * @param[in] remoteAddress         - the IP address of the session
+ *                                    initiator.
+ * @param[in] cleanupFn             - the callback to cleanup session on
+ *                                    destroy.
+ * @param[out] SessionIdentifier    - unique session ID
+ *
+ * @return int                      - session build status
+ */
+extern int obmcsesCreateWithFCleanup(const std::string& userName,
+                                     uint32_t remoteAddress,
```

Comment:
Seems I missed it in the previous review.
What about IPv6 addresses?

Reply:
Hmm, is there should be boost:asio::ip:address? I've very dislaked include boost into the current library only to define ip-address. Perhaps i can define new struct to store IPv4/IPv6 (value,type)?

Reply:
Are you sure that is should be stored in binary format?
Can we store it as a string?

Reply:
Perhaps You are right. I've preferred to use trivial types where possible, and the numeric comparison is faster, but it would be easier to store address as a string now.